src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -441,6 +441,9 @@
   <data name="ERR_BadBinaryOps" xml:space="preserve">
     <value>Operator '{0}' cannot be applied to operands of type '{1}' and '{2}'</value>
   </data>
+  <data name="ERR_AmbigBinaryOpsOnUnconstrainedDefault" xml:space="preserve">
+    <value>Operator '{0}' cannot be applied to operands of type '{1}' and '{2}' (this was incorrectly allowed in an early version of the compiler as an object comparison)</value>
```

Comment:
:memo: I'm open to better wording/message

Reply:
How about

> Operator '{0}' cannot be applied to unconstrained type parameters

Reply:
"unconstrained" can mean different things (no constraint, only constrained to an interface, constrained to another unconstrained type parameter). That seems a can of worms... :-(

Reply:
> "unconstrained" can mean different things (no constraint, only constrained to an interface, constrained to another unconstrained type parameter). That seems a can of worms... :-(

Not known to be a reference type? Or something similar,

---
In reply to: [364490884](https://github.com/dotnet/roslyn/pull/40801#discussion_r364490884) [](ancestors = 364490884)